be/src/olap/tablet_manager.cpp

```diff
@@ -533,11 +551,12 @@ OLAPStatus TabletManager::_drop_tablet_unlocked(
 OLAPStatus TabletManager::drop_tablets_on_error_root_path(
         const vector<TabletInfo>& tablet_info_vec) {
     OLAPStatus res = OLAP_SUCCESS;
-    WriteLock wlock(&_tablet_map_lock);
 
     for (const TabletInfo& tablet_info : tablet_info_vec) {
         TTabletId tablet_id = tablet_info.tablet_id;
         TSchemaHash schema_hash = tablet_info.schema_hash;
+        RWMutex& tablet_map_lock = _get_tablet_map_lock(tablet_id);
+        WriteLock wlock(&tablet_map_lock);
```

Comment:
When there are lots of tablet in `tablet_info_vec`, there will be many lock/unlock operation. Does  it matter?

Reply:
ok, I will refactor the code

Reply:
I think it may slow the clear task, but it is for the clear task, which is not so important for the performance, but there still something to do to reduce times of the  lock and unlock operation